src/algorithms/derivatives/hamiltonian_derivatives.jl

```diff
@@ -303,9 +314,9 @@ function (H::JordanMPO_AC_Hamiltonian)(x::MPSTensor)
     ismissing(H.C) || @plansor y[-1 -2; -3] += x[-1 2; 1] * H.C[-2 -3; 2 1]
     ismissing(H.B) || @plansor y[-1 -2; -3] += H.B[-1 -2; 1 2] * x[1 2; -3]
 
-    GL, A, GR = H.A
-    if nonzero_length(A) > 0
-        @plansor y[-1 -2; -3] += GL[-1 5; 4] * x[4 2; 1] * A[5 -2; 2 3] * GR[1 3; -3]
+    if !ismissing(H.A)
+        GLW, GR = H.A
+        @tensor y[-1 -2; -3] += GLW[-1 -2 3; 1 2] * x[1 2; 4] * GR[3 4; -3]
```

Comment:
I think here there is an interesting difference between planar and non-planar implementations.
I think your current implementation is optimal, unfortunately it is not planar, so will break our fermionic and anyonic code. The reason is that in planar mode we only allow cyclic permutations, which is not the case in the intermediate:
`GLWx[-1 -2 3; 4] * GR[3 4; -3]` has to cross legs 3 and 4 (which is precisely the uncrossing of the permutation you did in the constructor).
The interesting part is that we have to permute `GLWx` anyways, to bring it back into block-matrix form, but from a first guess this order is actually more beneficial... We can discuss tomorrow a bit more if you like.

TLDR, I think it is probably easier for now to keep `GR` unpermuted, and therefore also change the contraction to:
 
```suggestion
        @plansor y[-1 -2; -3] += GLW[-1 -2 3; 1 2] * x[1 2; 4] * GR[4 3; -3]
```